### DISCUSSION TITLE
how to get auth.user in model

### DISCUSSION THREAD

**Discussion by hariprasath-yadav** [adonisjs/core#3642] [Help] [answered]
How to get authenticated user details like id in Hooks functions (ex:  @beforeSave()) under database model

**Comment by Julien-R44** [comment, marked as answer]
Hello 👋 

As far as I know, since authentication depends on the context of the HTTP request, the only way to do this is to use local storage async. So basically, you have to enable local storage async: 
https://docs.adonisjs.com/guides/async-local-storage#usage

And then in your hook you can do :
```ts 
@beforeSave()
public static async myHook(user: User) {
    const auth = HttpContext.get()!.auth
   // And here you do what you want with the `auth` object.
}
```
